src/h5web/visualizations/index.tsx

```diff
@@ -1,101 +1,152 @@
-    },
-    defaultMappingState: (dataset) => {
-      const { dims } = dataset.shape as HDF5SimpleShape;
-      return dims.length === 1
-        ? ['x']
-        : [...range(dims.length - 2).fill(0), 'y', 'x'];
+    Component: MatrixVisContainer,
+    supportsEntity: (entity) => {
+      return (
+        isDataset(entity) &&
+        isBaseType(entity.type) &&
+        isSimpleShape(entity.shape) &&
+        entity.shape.dims.length >= 1
+      );
     },
   },
 
   [Vis.Line]: {
     Icon: FiActivity,
     Toolbar: LineToolbar,
-    Component: MappedLineVis,
-    supportsDataset: (dataset) => {
-      const { type, shape } = dataset;
+    Component: LineVisContainer,
+    supportsEntity: (entity) => {
       return (
-        isNumericType(type) && isSimpleShape(shape) && shape.dims.length >= 1
+        isDataset(entity) &&
+        isNumericType(entity.type) &&
+        isSimpleShape(entity.shape) &&
+        entity.shape.dims.length >= 1
       );
     },
-    defaultMappingState: (dataset) => {
-      const { dims } = dataset.shape as HDF5SimpleShape;
-      return [...range(dims.length - 1).fill(0), 'x'];
-    },
   },
 
   [Vis.Heatmap]: {
     Icon: FiMap,
     Toolbar: HeatmapToolbar,
-    Component: MappedHeatmapVis,
-    supportsDataset: (dataset) => {
-      const { type, shape } = dataset;
+    Component: HeatmapVisContainer,
+    supportsEntity: (entity) => {
+      return (
+        isDataset(entity) &&
+        isNumericType(entity.type) &&
+        isSimpleShape(entity.shape) &&
+        entity.shape.dims.length >= 2
+      );
+    },
+  },
+
+  [Vis.NxSpectrum]: {
+    Icon: FiActivity,
+    Toolbar: LineToolbar,
+    Component: NxSpectrumContainer,
+    supportsEntity: (entity, metadata) => {
+      if (!isGroup(entity) || !isNxDataGroup(entity)) {
+        return false;
+      }
+
+      const dataset = getSignalDataset(entity, metadata);
+      if (!dataset) {
+        return false;
+      }
+
+      const interpretation = getAttributeValue(dataset, 'interpretation');
+      if (interpretation) {
+        return interpretation === NxInterpretation.Spectrum;
+      }
+
       return (
-        isNumericType(type) && isSimpleShape(shape) && shape.dims.length >= 2
+        isNumericType(dataset.type) &&
+        isSimpleShape(dataset.shape) &&
+        dataset.shape.dims.length === 1
       );
     },
-    defaultMappingState: (dataset) => {
-      const { dims } = dataset.shape as HDF5SimpleShape;
-      return [...range(dims.length - 2).fill(0), 'y', 'x'];
+  },
+
+  [Vis.NxImage]: {
+    Icon: FiMap,
+    Toolbar: HeatmapToolbar,
+    Component: NxImageContainer,
+    supportsEntity: (entity, metadata) => {
+      if (!isGroup(entity) || !isNxDataGroup(entity)) {
+        return false;
+      }
+
+      const dataset = getSignalDataset(entity, metadata);
+      if (!dataset) {
+        return false;
+      }
+
+      const interpretation = getAttributeValue(dataset, 'interpretation');
```

Comment:
`interpretation` should be checked on `entity` not on `dataset`. Same for `NXSpectrum`

Reply:
The `entity` is the NxData group. According to the spec (and to the water file), the `interpretation` attribute is not defined on this group (https://manual.nexusformat.org/classes/base_classes/NXdata.html?highlight=nxdata) but on the `signal` _field_ (i.e. the `dataset`) that the NXdata group references: https://manual.nexusformat.org/nxdl_desc.html?highlight=interpretation#interpretation

Reply:
I will never get these right :joy_cat: